core/src/read.jl

```diff
@@ -230,6 +232,57 @@ function initialize_allocation!(p::Parameters, config::Config)::Nothing
     return nothing
 end
 
+"""
+Constructor for the ParameterUpdate object. Scalar Float64 fields have default value
+NaN and ScalarInterpolation values have default value LinearInterpolation(Float64[], Float64[])
+(an empty interpolation object). These default values get ignored in DiscreteControl.
+Non-supported key word inputs do not raise an error but get ignored via kwargs....
+"""
+function ParameterUpdate(
+    node_type::NodeType.T,
```

Comment:
Might I suggest some bedtime reading material? https://docs.julialang.org/en/v1/manual/types/#Parametric-Types

I think reading that might trigger a slight refactor of this struct and related code.

Reply:
I know about parametric types, that could at least resolve the `flow_rate` problem. I did actually make a parametric version of the `ParameterUpdate` struct at some point, but eventually I wanted it to be non-parametric so that I could put all their instances into a single type stable dictionary as mentioned above. I realize now that that is not needed for a type stable implementation as we can access a node type specific object containing the parameters of a control state  within the corresponding `update_parameters!` method. 

In response to your comment that `ParameterUpdate` is too uniony: I agree. I actually tried using the same field of `ParameterUpdate` for different variables for different node types, but that makes the code unnecessarily hard to understand. So the better solution is to make a `ParameterUpdate` struct for each node type. 

The point of indicating that not all controllable parameters are part of the control state remains, or we have to rethink that possibility. We could also say that unspecified optional parameters in a control state always get their default value, I'm not sure how that works in the code at the moment actually.